density_scaler/src/hex.rs

```diff
@@ -0,0 +1,300 @@
+use h3ron::{FromH3Index, H3Cell, Index};
+use itertools::Itertools;
+use lazy_static::lazy_static;
+use serde::Serialize;
+use std::{cmp, collections::HashMap, ops::Range};
+
+pub struct HexResConfig {
+    pub neighbors: u64,
+    pub target: u64,
+    pub max: u64,
+}
+
+impl HexResConfig {
+    pub fn new(neighbors: u64, target: u64, max: u64) -> Self {
+        Self {
+            neighbors,
+            target,
+            max,
+        }
+    }
+}
+
+type HexMap = HashMap<H3Cell, u64>;
+
+const DENSITY_TGT_RES: u8 = 4;
+const MAX_RES: u8 = 11;
+const USED_RES: Range<u8> = DENSITY_TGT_RES..MAX_RES;
+const SCALING_RES: Range<u8> = DENSITY_TGT_RES..MAX_RES + 2;
+
+lazy_static! {
+    static ref HIP17_RES_CONFIG: HashMap<u8, HexResConfig> = {
+        let mut conf_map = HashMap::new();
+        // Hex resolutions 0 - 3 and 11 and 12 are currently ignored by the
+        // global hex population constructor, filtered by their target value.
+        // For completeness sake their on-chain settings are N=2, TGT=100_000, MAX=100_000
+        conf_map.insert(4, HexResConfig::new(1, 250, 800));
+        conf_map.insert(5, HexResConfig::new(1, 100, 400));
+        conf_map.insert(6, HexResConfig::new(1, 25, 100));
+        conf_map.insert(7, HexResConfig::new(2, 5, 20));
+        conf_map.insert(8, HexResConfig::new(2, 1, 4));
+        conf_map.insert(9, HexResConfig::new(2, 1, 2));
+        conf_map.insert(10, HexResConfig::new(2, 1, 1));
+        conf_map
+    };
+}
```

Comment:
A static hashmap here may be a overkill. An alternative would be an array:

```suggestion
static HIP17_RES_CONFIG: [Option<HexResConfig>; 11] = [
    None,                                 // 0
    None,                                 // 1
    None,                                 // 2
    None,                                 // 3
    Some(HexResConfig::new(1, 250, 800)), // 4
    Some(HexResConfig::new(1, 100, 400)), // 5
    Some(HexResConfig::new(1, 25, 100)),  // 6
    Some(HexResConfig::new(2, 5, 20)),    // 7
    Some(HexResConfig::new(2, 1, 4)),     // 8
    Some(HexResConfig::new(2, 1, 2)),     // 9
    Some(HexResConfig::new(2, 1, 1)),     // 10
];
```

Reply:
i like this better for the use; TIL `pub const fn`